stdlib/public/Reflection/TypeLowering.cpp

```diff
@@ -1260,7 +1304,7 @@ class EnumTypeInfoBuilder {
         Size += tagCounts.numTagBytes;
         // Dynamic multi-payload enums use the tag representations not assigned
         // to cases for extra inhabitants.
-        if (tagCounts.numTagBytes >= 32) {
+        if (tagCounts.numTagBytes >= 4) {
```

Comment:
Unrelated typo in the original code that I thought I would fix while I'm here.

Reply:
Is there a possible test case?

Reply:
I don't think you can practically test this without constructing type information by hand specifically to exercise this path.  I believe it doesn't cause any problems until you have > 2^31 cases.